inc/ProviderRegistry.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Provider registry
+ *
+ * @package asset-manager-framework
+ */
+
+declare( strict_types=1 );
+
+namespace AssetManagerFramework;
+
+final class ProviderRegistry {
+
+	private static $instance;
+
+	private $providers = [];
+
+	private function __construct() {}
+
+	public static function instance(): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	public function register( Provider $provider ) : ProviderRegistry {
+		$provider = apply_filters( 'amf/provider', $provider );
```

Comment:
This feels like perhaps the most significant change but I guess the results will be the same with the filter only needing to run once rather than on every get. Is that the main reason for the switch?

Reply:
Ahh sorry, I think that was @shadyvb. I thought I removed it as within the function it'd create a bug where you'd get whatever the last returned provider was rather than a specific provider.

Reply:
So, is the filter inside `register` good then? 🙂

Reply:
Yeah I think it's fine, doesn't have the same issue, resolving this convo :)